recipes/opentdf-client/all/conanfile.py

```diff
@@ -36,63 +39,105 @@ def _minimum_cpp_standard(self):
             "Visual Studio": "17",
+            "msvc": "19.22",
             "gcc": "7.5.0",
             "clang": "12",
             "apple-clang": "12.0.0",
         }
 
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for data in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(data["patch_file"])
+
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+            check_min_cppstd(self, self._minimum_cpp_standard)
         min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
         if not min_version:
-            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
-                self.name, self.settings.compiler))
+            self.output.warn(f"{self.name} recipe lacks information about the {self.settings.compiler} compiler support.")
         else:
-            if tools.Version(self.settings.compiler.version) < min_version:
-                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
-                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
+            if Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration(f"{self.name} requires C++{self._minimum_cpp_standard} support."
+                                                 "The current compiler {self.settings.compiler} {self.settings.compiler.version} does not support it.")
 
     def requirements(self):
-        self.requires("openssl/1.1.1l@")
-        self.requires("boost/1.76.0@")
-        self.requires("zlib/1.2.11@")
-        self.requires("ms-gsl/2.1.0@")
-        self.requires("libxml2/2.9.10@")
-        self.requires("libarchive/3.5.1@")
-        self.requires("nlohmann_json/3.10.4@")
-        self.requires("jwt-cpp/0.4.0@")
+        self.requires("openssl/1.1.1q")
+        self.requires("boost/1.79.0")
+        self.requires("ms-gsl/2.1.0")
+        self.requires("libxml2/2.9.14")
+        self.requires("nlohmann_json/3.11.1")
+        self.requires("jwt-cpp/0.4.0")
+        if self.options.get_safe("with_libarchive"):
+            self.requires("libarchive/3.6.1")
+        # We do not require libiconv but conan-center only allows 'stock' references, and boost+libxml2
+        # specify differerent versions, which causes a build fail due to the dependency conflict.
+        # Overriding the version here allows a clean build with the stock build settings.
+        # Preferred solution is to set with_libiconv to False
+        if self.options.get_safe("with_libiconv"):
+            self.requires("libiconv/1.17@")
+        # Same for zlib
+        # Preferred solution is to set with_zlib to False
+        if self.options.get_safe("with_zlib"):
+            self.requires("zlib/1.2.12@")
```

Comment:
```suggestion
```
Boost 1.79.0 has been updated.

Reply:
Updated

Reply:
Q:  I added the 'branch_version" method of getting sources to facilitate developing other applications that we have that are dependent on pre-release code changes.  

Without that option, I have to do a full release of opentdf-client and have it published to conan-center (2 days at best using automated process, 2 weeks at best for a manual update like this) before I can start working on an application that consumes the new functionality.  If that new functionality doesn't meet the needs, I have to repeat the cycle again and do another release before I can try again.

With that option, I can create a branch in opentdf-client, and have the application consume `opentdf-client/branch-name@' and do whatever iterative development is necessary before merging to main and cutting a release.  That way everything can be resolved in one release cycle.

Reply:
@uilianries 

As for your PRs...I have to admit that I did not see them come in, my sincere apologies.  Looks like I need to update my notification prefs so I don't miss anything in the future.  Thank you so much for contributing, it is much appreciated!

I had a PR in progress to include your changes, looks like I don't need it now.

Reply:
Conan Center Index does not count with local development. Unfortunately, Conan Development process does not work very well for Conan Center Index and Local development as well. All recipes in this repository follow that order, only download releases and does take local development into account.

In the very past I tried to combine [Package Dev Flow](https://docs.conan.io/en/latest/developing_packages/package_dev_flow.html) with Conan Center Index, but didn't work COnan v1 limitations. We can try to do something with Conan v2. 

On the other hand, there is a feature request which could fit your case: https://github.com/conan-io/conan/issues/6453 As you want local development, you could use local tar file without any new release.

Reply:
Updated...